### ISSUE TITLE
Bad output?

### ISSUE THREAD

**Issue by barbeau** [barbeau/earth-gravitational-model#1]
While this project passes the one unit test bundled with Geotools for the EGM class, I haven't gotten the output to match some online tools given lat/longs in the Tampa, FL area.  So, I'm not entirely convinced that this project is behaving correctly - more unit tests are needed.

One potential source for sample unit test data is this project:
https://github.com/matthiaszimmermann/EGM96/blob/master/src/test/java/org/matthiaszimmermann/location/egm96/GeoidTest.java#L145

Or just mine it from existing EGM conversion tools online.

**Comment by barbeau** [comment]
Here's a branch with the unit tests from https://github.com/matthiaszimmermann/EGM96/blob/master/src/test/java/org/matthiaszimmermann/location/egm96/GeoidTest.java#L145 added - it's failing those tests:
https://github.com/barbeau/earth-gravitational-model/tree/egm96Test

Commit with new tests:
https://github.com/barbeau/earth-gravitational-model/commit/be62080e35ecf36e6866ca2fb47d5cabe5083156

**Comment by benlc** [comment]
I also am observing wrong output using the provided code and EGM database.

**Comment by barbeau** [comment]
From https://github.com/opentripplanner/OpenTripPlanner/issues/2301#issuecomment-280490788:

>Note that in the course of writing/testing this, I discovered that there are several different geoids that have been associated with WGS84. As best I can tell, the egm180.nor data (which the GeoTools EarthGravitationalModel is based on) refers to the EGM84 geoid, rather than the newer EGM96 and EGM2008 models which some of the above utilities use. Here is a handy tool for calculating all three at once: http://geographiclib.sourceforge.net/cgi-bin/GeoidEval
>
>I tested EarthGravitationalModel as used in this OTP branch for about a dozen places around the world and got exact matches with EGM84 values from that calculator. @barbeau -- the unit tests you referenced appear to be written for EGM96, which is perhaps why you saw some failures in your port of the Geotools class. Also, those tests appear to use (0-360) longitudes, whereas I am using (-180-180).

I haven't had a chance to test this myself yet.

@benlc Are you sure you're testing EGM84 values?

**Comment by barbeau** [comment]
I've looked at this more closely, and it appears that the egm180.nor file does indeed include coefficients for the EGM84 model, and that's why the other unit tests based on EGM96 do not match. I've confirmed this using real-world values from GPSTest as well, and added another unit test based on this here - https://github.com/barbeau/earth-gravitational-model/commit/976db41d0c8567dc0b7405dace91e8046c60974f. These observed values match EGM84 values from http://geographiclib.sourceforge.net/cgi-bin/GeoidEval.
